File: aspn-py/src/aspn23/type_satnav_satellite_system.py
"""
This code is generated via firehose.
DO NOT hand edit code.  Make any changes required using the firehose repo instead.
"""

from dataclasses import dataclass
from enum import Enum
from typing import Optional

from numpy.typing import NDArray


class TypeSatnavSatelliteSystemSatelliteSystem(Enum):

    """
    Enumerated field which describes the Satellite System.
    """

    """
    GPS.
    """
    SYS_GPS = 0

    """
    Galileo.
    """
    SYS_GALILEO = 1

    """
    GLONASS.
    """
    SYS_GLONASS = 2

    """
    BeiDou.
    """
    SYS_BEIDOU = 3

    """
    SBAS.
    """
    SYS_SBAS = 4

    """
    QZSS.
    """
    SYS_QZSS = 5

    """
    IRNSS.
    """
    SYS_IRNSS = 6


@dataclass
class TypeSatnavSatelliteSystem:
    """
    Satellite system name

    ### Attributes

    satellite_system - TypeSatnavSatelliteSystemSatelliteSystem:
            Enumerated field which describes the Satellite System.
    """

    satellite_system: TypeSatnavSatelliteSystemSatelliteSystem

File: aspn-c/src/aspn23/TypeSatnavSatelliteSystem.c

/*
 * This code is generated via firehose.
 * DO NOT hand edit code.  Make any changes required using the firehose repo instead
 */

#include "TypeSatnavSatelliteSystem.h"

Aspn23TypeSatnavSatelliteSystem* ASPN_NULLABLE aspn23_type_satnav_satellite_system_new(
    enum Aspn23TypeSatnavSatelliteSystemSatelliteSystem satellite_system) {
	Aspn23TypeSatnavSatelliteSystem* self =
	    (struct Aspn23TypeSatnavSatelliteSystem*)malloc(sizeof(Aspn23TypeSatnavSatelliteSystem));
	if (NULL == self) return NULL;
	self->satellite_system = satellite_system;
	return self;
}

Aspn23TypeSatnavSatelliteSystem* ASPN_NULLABLE
aspn23_type_satnav_satellite_system_copy(Aspn23TypeSatnavSatelliteSystem* input) {

	Aspn23TypeSatnavSatelliteSystem* out =
	    aspn23_type_satnav_satellite_system_new(input->satellite_system);

	return out;
}

void aspn23_type_satnav_satellite_system_free(void* pointer) {
	Aspn23TypeSatnavSatelliteSystem* self = (Aspn23TypeSatnavSatelliteSystem*)pointer;
	if (NULL == self) return;
	aspn23_type_satnav_satellite_system_free_members(self);
	free(self);
}

void aspn23_type_satnav_satellite_system_free_members(Aspn23TypeSatnavSatelliteSystem* self) {
	if (NULL == self) return;
}
